dotCMS/src/main/java/com/dotcms/contenttype/model/field/FieldTypeFactory.java

```diff
@@ -0,0 +1,74 @@
+package com.dotcms.contenttype.model.field;
+
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.language.LanguageException;
+import com.liferay.portal.language.LanguageUtil;
+import com.liferay.portal.model.User;
+
+import java.util.*;
+
+/**
+ * Created by freddyrodriguez on 7/27/17.
+ */
+public class FieldTypeFactory {
+
+    private Map<String, Field> fieldsMap;
+
+    private static class SingletonHolder {
+        private static final FieldTypeFactory INSTANCE = new FieldTypeFactory();
+    }
+
+    public static FieldTypeFactory getInstance() {
+
+        return FieldTypeFactory.SingletonHolder.INSTANCE;
+    }
+
+    private FieldTypeFactory(){
+        fieldsMap = new HashMap<>();
+        LegacyFieldTypes[] legacyFieldTypes = LegacyFieldTypes.values();
+
+        for (LegacyFieldTypes legacyFieldType : legacyFieldTypes) {
+            String id = legacyFieldType.legacyValue();
+
+            Class<? extends Field> fieldClass = legacyFieldType.implClass();
+            addField(id, fieldClass);
+        }
+    }
+
+    public void addField(String id, Class<? extends Field> fieldClass) {
+        fieldsMap.put( id, createFieldIntance( fieldClass ) );
+    }
+
+    public Collection<FieldType> getFieldTypes(User user){
+        Collection<FieldType> fieldTypes = new TreeSet();
+
+        for (Map.Entry<String, Field> fieldEntry : fieldsMap.entrySet()) {
+            String id = fieldEntry.getKey();
+            Field field = fieldEntry.getValue();
+
+            Class<? extends Field> fieldClass = field.getClass();
+
+            try {
+                String label = LanguageUtil.get( user, field.getContentTypeFieldLabelKey() );
+                String helpText = UtilMethods.escapeSingleQuotes( LanguageUtil.get( user, field.getContentTypeFieldHelpTextKey() ) );
+                Collection<ContentTypeFieldProperties> fieldContentTypeProperties = field.getFieldContentTypeProperties();
+
+                if (fieldContentTypeProperties != null) {
+                    FieldType fieldType = new FieldType(id, label, fieldContentTypeProperties, helpText, fieldClass.getName());
+                    fieldTypes.add(fieldType);
+                }
+            } catch (LanguageException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)